frontend/src/metabase/static-viz/components/LineAreaBarChart/utils/series.unit.spec.ts

```diff
@@ -0,0 +1,401 @@
+      [
+        {
+          name: "Count",
+          cardName: "Bar chart",
+          yAxisPosition: "left",
+          type: "bar",
+          data: [
+            ["Doohickey", 3976],
+            ["Gadget", 4939],
+            ["Gizmo", 4784],
+            ["Widget", 5061],
+          ],
+          column: {
+            name: "count",
+            source: "aggregation",
+            display_name: "Count",
+          },
+        },
+      ],
+    ];
+
+    it("should assign colors given series", () => {
+      const seriesWithColors = getSeriesWithColors(
+        multipleSeries,
+        settings,
+        getPalette({}),
+      );
+
+      const expectedSeries = [
+        [
+          {
+            color: "#509EE3", // brand color
+            name: expect.anything(),
+            cardName: expect.anything(),
+            yAxisPosition: expect.anything(),
+            type: expect.anything(),
+            data: expect.anything(),
+            column: expect.anything(),
+          },
+        ],
+      ];
+
+      expect(seriesWithColors).toEqual(expectedSeries);
+    });
+
+    it("should assign colors from whitelabel colors", () => {
+      const seriesWithColors = getSeriesWithColors(
+        multipleSeries,
+        settings,
+        getPalette({ brand: "#123456", summarize: "#ffffff" }),
+      );
+
+      const expectedSeries = [
+        [
+          {
+            color: "#123456", // whitelabel color
+            name: expect.anything(),
+            cardName: expect.anything(),
+            yAxisPosition: expect.anything(),
+            type: expect.anything(),
+            data: expect.anything(),
+            column: expect.anything(),
+          },
+        ],
+      ];
+
+      expect(seriesWithColors).toEqual(expectedSeries);
+    });
+
+    it("it should assign colors from column colors", () => {
+      const seriesWithColors = getSeriesWithColors(
+        multipleSeries,
+        merge(settings, {
+          series_settings: {
+            count: {
+              color: "#987654",
+            },
+          },
+        }),
+        getPalette({ brand: "#123456", summarize: "#ffffff" }),
+      );
+
+      const expectedSeries = [
+        [
+          {
+            color: "#987654", // column color
+            name: expect.anything(),
+            cardName: expect.anything(),
+            yAxisPosition: expect.anything(),
+            type: expect.anything(),
+            data: expect.anything(),
+            column: expect.anything(),
+          },
+        ],
+      ];
+
+      expect(seriesWithColors).toEqual(expectedSeries);
+    });
+  });
+
+  describe("Series with preferred colors", () => {
+    const multipleSeries: SeriesWithOneOrLessDimensions[][] = [
+      [
+        {
+          name: "Sum of Total",
+          cardName: "Bar chart",
+          yAxisPosition: "left",
+          type: "bar",
+          data: [["2016-04-24T00:00:00Z", 52.75594257942132]],
+          column: {
+            name: "sum",
```

Comment:
for a series key (in this particular case, `_seriesKey` is from `column.name`) with value that's defined as one of the option in this function https://github.com/metabase/metabase/blob/a4255c5e749bfc3053f20d3c64010055ae349917/frontend/src/metabase/lib/colors/groups.ts#L41

it would get a predetermined color